concept/type_/annotation.rs

```diff
@@ -82,11 +105,118 @@ impl AnnotationRegex {
         Self { regex }
     }
 
+    pub fn default() -> Self {
+        Self::new(".*".to_owned())
+    }
+
     pub fn regex(&self) -> &str {
         &self.regex
     }
 }
 
+impl Annotation {
+    pub fn category(&self) -> AnnotationCategory {
+        match self {
+            Self::Abstract(_) => AnnotationCategory::Abstract,
+            Self::Distinct(_) => AnnotationCategory::Distinct,
+            Self::Independent(_) => AnnotationCategory::Independent,
+            Self::Unique(_) => AnnotationCategory::Unique,
+            Self::Key(_) => AnnotationCategory::Key,
+            Self::Cardinality(_) => AnnotationCategory::Cardinality,
+            Self::Regex(_) => AnnotationCategory::Regex,
+            Self::Cascade(_) => AnnotationCategory::Cascade,
+        }
+    }
+}
+
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct AnnotationCascade;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum AnnotationCategory {
+    Abstract,
+    Distinct,
+    Independent,
+    Unique,
+    Key,
+    Cardinality,
+    Regex,
+    Cascade,
+    // TODO: Subkey
+    // TODO: Values
+    // TODO: Range
+    // TODO: Replace
+}
+
+impl AnnotationCategory {
+    pub fn to_default_annotation(&self) -> Annotation {
+        // TODO: fn const for regex default
```

Comment:
Wanted to make this function const, but could not make `AnnotationRegex::default()` const because of the `String`s things. Might go back to try again